Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs

```diff
@@ -1007,35 +1012,36 @@ private void DestroyAllServices()
 
         private bool ExecuteOnAllServicesInOrder(Action<IMixedRealityService> execute)
         {
-            if (!HasProfileAndIsInitialized)
-            {
-                return false;
-            }
-
-            var services = MixedRealityServiceRegistry.GetAllServices();
-            int length = services.Count;
-            for (int i = 0; i < length; i++)
-            {
-                execute(services[i]);
-            }
-
-            return true;
+            return ExecuteOnServiceCollection(
+                execute,
+                MixedRealityServiceRegistry.GetAllServices());
         }
 
         private bool ExecuteOnAllServicesReverseOrder(Action<IMixedRealityService> execute)
+        {
+            return ExecuteOnServiceCollection(
+                execute,
+                MixedRealityServiceRegistry.GetAllServices().Reverse());
```

Comment:
So oddly enough and I wish we caught this earlier, but GetAllServices() actually returns a HashSet which is not actually sorted in a deterministic order:

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/f7494ca8b3b8331d3fa184caf970be8d18acc6d5/Assets/MixedRealityToolkit/Inspectors/MixedRealityToolkitFacadeHandler.cs#L91

https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.hashset-1?view=netframework-4.8#remarks

I guess this works as-is just because we're lucky but I guess we could still get bit by this going forward...

@Troy-Ferrell  FYI

Reply:
@wiwei I don't think this is applicable? The linked cs file you have is the FacadeHandler. 

MRTK.cs uses MixedRealityServiceRegistry which sorts the services on registeration.

Do we care if the FacadeHandler renders/handles services in any order?